src/components/Bounty/BountyEntryFilesModal.tsx

```diff
@@ -0,0 +1,141 @@
+import {
+  Anchor,
+  Center,
+  Group,
+  Loader,
+  Paper,
+  ScrollArea,
+  Stack,
+  Text,
+  ThemeIcon,
+  Title,
+  Tooltip,
+} from '@mantine/core';
+import { Currency } from '@prisma/client';
+import { IconLock, IconLockOpen, IconStar } from '@tabler/icons-react';
+import React, { forwardRef } from 'react';
+import { createContextModal } from '~/components/Modals/utils/createContextModal';
+import { trpc } from '~/utils/trpc';
+import { BountyEntryGetById } from '~/types/router';
+import { formatKBytes } from '~/utils/number-helpers';
+import { CurrencyBadge } from '~/components/Currency/CurrencyBadge';
+
+type Props = { bountyEntry: Omit<BountyEntryGetById, 'files'> };
+
+const { openModal: openBountyEntryFilesModal, Modal } = createContextModal<Props>({
+  name: 'bountyEntryFiles',
+  title: 'Files',
+  size: 'md',
+  Element: ({ props }) => {
+    return <BountyEntryFiles {...props} />;
+  },
+});
+
+export { openBountyEntryFilesModal };
+export default Modal;
+
+function BountyEntryFiles({ bountyEntry }: Props) {
+  const { data: files, isLoading } = trpc.bountyEntry.getFiles.useQuery({ id: bountyEntry.id });
+
+  if (isLoading) {
+    return (
+      <Center>
+        <Loader />
+      </Center>
+    );
+  }
+
+  if (!files?.length) {
+    return (
+      <Stack>
+        <Text>Looks like this entry contains no files.</Text>
+      </Stack>
+    );
+  }
+
+  const hasLockedFiles = files.find(
+    (file) => (file.metadata.unlockAmount ?? 0) > bountyEntry.awardedUnitAmountTotal
+  );
+
+  return (
+    <Stack>
+      {hasLockedFiles && (
+        <Text>
+          Some of the files in this entry are still because they have not reached the unlock amount.
```

Comment:
Missed a word there it seems, fix as you see fit
```suggestion
          Some of the files in this entry are still not available because they have not reached the unlock amount.
```